src/main/java/executor/service/controller/ScenarioSourceController.java

```diff
@@ -0,0 +1,80 @@
+package executor.service.controller;
+
+import executor.service.model.request.ScenarioRequest;
+import executor.service.model.response.ScenarioResultResponse;
+import executor.service.service.ScenarioResultLoggingService;
+import executor.service.service.SourceHandler;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+import static executor.service.Routes.SCENARIOS;
+
+/**
+ * The {@code ScenarioSourceController} class
+ * represents a REST controller responsible for handling scenario-related requests and results.
+ * <p>
+ *
+ * @author Oleksandr Tuleninov
+ * @version 01
+ * @see SourceHandler
+ * @see ScenarioResultLoggingService
+ * @see ScenarioRequest
+ * @see ScenarioResultResponse
+ */
+@RestController
+@RequestMapping(value = SCENARIOS)
+public class ScenarioSourceController {
+
+    private final SourceHandler handler;
+    private final ScenarioResultLoggingService scenarioResultLoggingService;
```

Comment:
2 different services per 1 controller is not good

Reply:
It is good idea. I did it. Commit 1cc61c2a.